### PR TITLE
Transcendentals2

### PR DESCRIPTION
**Pull request by adamnemecek**
Based on #551, adds sin and cos. Implementations are based on the boost implementation

https://www.boost.org/doc/libs/1_63_0/boost/math/quaternion.hpp

### CODE REVIEW

File: src/geometry/quaternion.rs [hunk 1]
```diff
@@ -21,6 +21,27 @@ use base::{Matrix3, MatrixN, MatrixSlice, MatrixSliceMut, Unit, Vector3, Vector4
 
 use geometry::Rotation;
 
+/// Calculates the cardinal sine function
+#[inline]
+fn sinc<N: Real>(v: N) -> N {
+    if v == N::zero() {
+        N::zero()
```
Comment (sebcrozet) [inline src/geometry/quaternion.rs]:
```suggestion
        N::one()
```

File: src/geometry/quaternion.rs [hunk 2]
```diff
@@ -21,6 +21,27 @@ use base::{Matrix3, MatrixN, MatrixSlice, MatrixSliceMut, Unit, Vector3, Vector4
...
+    }
+    else {
+        v.sin() / v
+    }
+}
+
+#[inline]
+fn sinhc<N: Real>(v: N) -> N {
+    if v == N::zero() {
+        N::zero()
```
Comment (sebcrozet) [inline src/geometry/quaternion.rs]:
```suggestion
        N::one()
```

File: src/geometry/quaternion.rs [hunk 3]
```diff
@@ -506,6 +527,22 @@ impl<N: Real> Quaternion<N> {
     pub fn normalize_mut(&mut self) -> N {
         self.coords.normalize_mut()
     }
+
+    /// Calculates quaternionic sin
```
Comment (sebcrozet) [inline src/geometry/quaternion.rs]:
```suggestion
    /// Calculates the quaternionic cosinus.
```

File: src/geometry/quaternion.rs [hunk 4]
```diff
@@ -506,6 +527,22 @@ impl<N: Real> Quaternion<N> {
...
+    #[inline]
+    pub fn cos(&self) -> Self {
+        let z = self.imag().magnitude();
+        let w = -self.w.sin() * sinhc(z);
+        Self::from_parts(self.w.cos() * z.cosh(), self.imag() * w)
+    }
+
+    /// Calculates quaternionic sin
```
Comment (sebcrozet) [inline src/geometry/quaternion.rs]:
```suggestion
    /// Calculates the quaternionic sinus.
```

Comment (tpdickso) [reply src/geometry/quaternion.rs]: I think "sine" and "cosine" are the most common spellings of these in English.

**Pull request closed without merging**